client/src/components/userprofiles/UserProfileDetails.jsx

```diff
@@ -29,7 +34,9 @@ export default function UserProfileDetails() {
         </div>
        <div className="ms-1">
       <img alt="user profile image" className="w-25 m-auto"
-      src={userProfile.imageLocation||defaultPic} style={{ borderRadius: "50%"}}/>
+         src={userProfile.imageBlob ? `data:image/jpeg;base64,${userProfile.imageBlob}` : (userProfile.imageLocation || defaultPic)} style={{ borderRadius: "50%"}}/>
```

Comment:
Could you put the closing angle-bracket onto a new line, and have it at the same indentation level as the img tag